### PR TITLE
spectral picker chart finished

### PR DESCRIPTION
**Pull request by jpakey99**

### CODE REVIEW

**Review by egc2846 — DISMISSED** [review]

**Review by sk4900 — APPROVED** [review]

**Comment by keenanm500** [comment]
I tested this, and I have to say the new color spectrum looks 🔥 

I do think there has to be a way to decide when we want to add a new line to the graph or completely replace the graph like it does today. There should at least be a way to remove lines from the graph. Let me know if you want to chat about this

File: frontend/src/renderer/components/Charts/LineChart.svelte [hunk 1]
```diff
@@ -1,86 +1,163 @@
...
+  let lineNumber = 1;
+  let seriesData = [];
 
   $: if (data.length > 1) {
-    inputData = [];
-    wavelengthArray.forEach((element, i) => {
-      inputData.push({
-        intensity: data[i] * 100,
-        wavelength: element,
+      let dataDict = [];
```
Comment (keenanm500) [inline frontend/src/renderer/components/Charts/LineChart.svelte:21]: this could be `const`

**Review by keenanm500 — CHANGES_REQUESTED** [review]

File: frontend/src/renderer/components/Charts/LineChart.svelte [hunk 2]
```diff
@@ -1,86 +1,163 @@
...
+          type: 'line',
+          stroke: {
+              curve: 'smooth',
+          },
+          toolbar: {
+              // Hamburger menu which has exports such as CSV etc.
+              // I have had issues displaying this, I believe some unrelated global CSS is causing issues
+              show: true,
+              tools: {
+                  download: true,
```
[the remark this answers is not in the document]
Comment (keenanm500) [reply frontend/src/renderer/components/Charts/LineChart.svelte:67]: I had difficulties with this also, we could check with them about how important it is. We can definitely figure this out though if necessary

**Review by Victoriam7 — DISMISSED** [review]

File: frontend/src/renderer/components/Charts/LineChart.svelte [hunk 3]
```diff
@@ -1,86 +1,163 @@
...
+          },
+          markers: {
+              fillColors: seriesColors,
+          }
+      },
+      title: {
+          text: "Estimated Spectrum",
+          align: 'left',
+          style: {
+              color:  '#fff'
```
[the remark this answers is not in the document]
Comment (sk4900) [reply frontend/src/renderer/components/Charts/LineChart.svelte:151]: I agree, I think we should use the full 6 characters in every case. Caps preferred

**Review by sk4900 — DISMISSED** [review]

File: frontend/src/renderer/components/Charts/LineChart.svelte [hunk 4]
```diff
@@ -1,86 +1,163 @@
...
   export let data = [];
-  export let wavelengthArray;
+  export let wavelengthArray = Array.from({ length: 35 }, (x, i) => i * 10 + 380);;
   export let trueShadowPos;
+  let seriesColors = ['#610061', '#79008d', '#8300b5', '#7e00db', '#6a00ff', '#3d00ff', '#0000ff', '#0046ff', '#007bff', '#00a9ff', '#00d5ff', '#00ffff', '#00ff92', '#00ff00', '#36ff00', '#5eff00', '#81ff00', '#a3ff00', '#c3ff00', '#ffff00', '#ffdf00', '#ffbe00', '#ff9b00', '#ff7700', '#ff4f00', '#ff2100', '#ff0000', '#ff0000', '#ff0000', '#ff0000', '#ff0000', '#ff0000', '#ff0000', '#ff0000', '#ff0000'];
+
 
   let inputData = [];
+  let lineNumber = 1;
+  let seriesData = [];
```
[the remark this answers is not in the document]
Comment (jpakey99) [reply frontend/src/renderer/components/Charts/LineChart.svelte:18]: nope.  I forgot to delete them from when I was trying something else.  Will make sure I delete

File: frontend/src/renderer/components/Charts/LineChart.svelte (hunk 3 above)
[the remark this answers is not in the document]
Comment (jpakey99) [reply frontend/src/renderer/components/Charts/LineChart.svelte:151]: sounds good.  I will make the change

File: frontend/src/renderer/components/Charts/LineChart.svelte [hunk 5]
```diff
@@ -119,163 +196,4 @@
     height: 0;
     padding: 0;
   }
-
```
[the remark this answers is not in the document]
Comment (jpakey99) [reply frontend/src/renderer/components/Charts/LineChart.svelte:122]: I am pretty sure everything I deleted did not work with apex and was specific for the old chart

**Review by egc2846 — APPROVED** [review]

File: frontend/src/renderer/components/Charts/LineChart.svelte (hunk 2 above)
[the remark this answers is not in the document]
Comment (keenanm500) [reply frontend/src/renderer/components/Charts/LineChart.svelte:65]: I'm sure there's a way to fix this. Might be worth pair-programming to find a fix. If there isn't a way to fix it normally, we could probably hack it by using an icon without text or something

[the remark this answers is not in the document]
Comment (keenanm500) [reply frontend/src/renderer/components/Charts/LineChart.svelte:72]: @jpakey99 Got this fixed, nice work

**Review by keenanm500 — APPROVED** [review]

**Review by sk4900 — APPROVED** [review]

**Pull request merged**